src/style-spec/expression/stops.js

```diff
@@ -11,19 +11,22 @@ export type Stops = Array<[number, Expression]>;
  * @private
  */
 export function findStopLessThanOrEqualTo(stops: Array<number>, input: number) {
-    const n = stops.length;
+    const n = stops.length - 1;
     let lowerIndex = 0;
-    let upperIndex = n - 1;
+    let upperIndex = n;
     let currentIndex = 0;
-    let currentValue, upperValue;
+    let currentValue, nextValue;
 
     while (lowerIndex <= upperIndex) {
         currentIndex = Math.floor((lowerIndex + upperIndex) / 2);
         currentValue = stops[currentIndex];
-        upperValue = stops[currentIndex + 1];
-        if (input === currentValue || input > currentValue && input < upperValue) { // Search complete
+        nextValue = stops[currentIndex + 1];
+
+        if (currentValue <= input && (currentIndex === n || input < nextValue)) { // Search complete
             return currentIndex;
-        } else if (currentValue < input) {
+        }
+
+        if (currentValue <= input) {
```

Comment:
Nit: can we put the "search complete" condition inside this `if` so that we don't check `currentValue <= input` twice?